zod/cli/torrent_download.py

```diff
@@ -0,0 +1,347 @@
+            files.append("vehicle_data")
+
+    # frame infos
+    if filter_settings.infos:
+        files.append("infos")
+
+    file_indicies = [files_to_idx[f] for f in files]
+
+    files = [f"{file}.tar.gz" for file in files]
+
+    return files, file_indicies
+
+
+def download_torrent_file(subset: SubDataset, output_dir: Path):
+    """Download the torrent file for the given subset and version."""
+    url = TORRENT_FILEMAP[subset]
+    filename = f"{subset.name.lower()}.torrent"
+    print("Downloading torrent file: {}".format(filename))
+    try:
+        urllib.request.urlretrieve(url, filename)
+        # move the torrent file to the output directory
+        os.rename(filename, output_dir / filename)
+
+        assert os.path.getsize(output_dir / filename) == SIZE_FILEMAP[subset], (
+            f"Downloaded torrent file is not the correct size. "
+            f"Expected: {SIZE_FILEMAP[subset]}, "
+            f"Actual: {os.path.getsize(output_dir / filename)}"
+        )
+        return output_dir / filename
+    except Exception as e:
+        print(f"Failed to download torrent file: {e}")
+        exit(1)
+
+
+def _download_dataset(
+    dl_settings: DownloadSettings, filter_settings: FilterSettings, subset: SubDataset
+):
+    dl_dir = Path(dl_settings.output_dir) / "downloads"
+    dl_dir.mkdir(parents=True, exist_ok=True)
+
+    torrent_filepath = download_torrent_file(subset, dl_dir)
+
+    files, file_indicies = get_files_to_download(subset, filter_settings)
+    # create the aria2c command
+    cmd = [
+        "aria2c",
+        f"--select-file={','.join([str(i) for i in file_indicies])}",
+        f"--torrent-file={torrent_filepath}",
+        f"--dir={str(dl_dir)}",
+        "--seed-time=0",
+        "--continue=true",
+    ]
+
+    if dl_settings.dry_run:
+        print(cmd)
+        cmd.append("--dry-run")
+
+    subprocess.check_call(cmd)
+
+    # aria2c will download adjacent files in the torrent, so we need to remove those files
+    # that we don't want
+    dl_dir = dl_dir / subset.name.lower()
+    for f in dl_dir.iterdir():
+        if f.name not in files:
+            f.unlink()
+
+    # if we are extracting, extract the tar gz files
+    if dl_settings.extract:
+        for f in dl_dir.iterdir():
+            if f.name.endswith(".tar.gz"):
+                with tarfile.open(f, "r:gz") as tar:
+                    for member in tqdm(
+                        iterable=tar.getmembers(),
+                        total=len(tar.getmembers()),
+                        desc=f"Extracting {f.name}",
+                    ):
+                        tar.extract(member=member, path=dl_settings.output_dir)
+
+    # if we are removing the archives, remove them
+    if dl_settings.rm:
+        shutil.rmtree(dl_dir)
+
+
+REQ = "Required Arguments"
+GEN = "General Download Settings"
+FIL = "General Filter Settings"
+FRA = "Frames Filter Settings"
+SEQ = "Sequences/Drives Filter Settings"
+
+
+@app.command()
+def download(
```

Comment:
almost everything here is shared with the dropbox download script. Can we have a single, shared, entrypoint to simplify maintenance?
something like:
zod download --torrent (with interactive question if not specified)
or hierarchical like 
zod download torrent vs zod download dropbox (but maybe messy to share arguments then?)

Reply:
I made it to be an argument in the download script, acc. your first suggestion. Now we have an entrypoint that chooses what downloading script to run based on user choice.